tinygrad/dtype.py

```diff
@@ -1,17 +1,18 @@
 from __future__ import annotations
-from typing import Final, Optional, ClassVar, Set, Tuple, Dict, Union, Callable
+from typing import Final, Optional, ClassVar, Set, Tuple, Dict, Union, Callable, TypeVar, Generic
 import math, struct, ctypes, functools
 from dataclasses import dataclass
 from tinygrad.helpers import getenv
 
 ConstType = Union[float, int, bool]
+DtypeFmtType = TypeVar('DtypeFmtType', bound=str)
 
 @dataclass(frozen=True, order=True)
-class DType:
+class DType(Generic[DtypeFmtType]):
```

Comment:
Why?

Reply:
This was the cleanest way I found to fix the type issues without adding a bunch of asserts in different places. Anyway, I removed all of these changes.